pallets/parachain-staking/src/lib.rs

```diff
@@ -975,6 +975,9 @@ pub mod pallet {
 					);
 
 					if remaining.is_zero() {
+						// we do not remove the scheduled delegation requests from other collators
+						// since it is assumed that they were removed incrementally before only the
+						// last delegation was left.
```

Comment:
thanks for adding this comment

the assumption that is **if no other delegations exist, then no other delegation requests exist**